src/mobile_crawler/domain/video_recording_manager.py

```
@@ -276,21 +249,85 @@ def save_partial_on_crash(self) -> Optional[str]:
+                return None
+            
+            device_id = self.appium_driver.device_id
+            
+            # List mp4 files in /sdcard/, sorted by time (newest first)
+            # screenrecord files usually don't have a standard name if initiated by Appium 
+            # (Appium often uses a random UUID), but we can guess.
+            import subprocess
+            
+            # Find recent mp4 files
+            cmd = ['adb', '-s', device_id, 'shell', 'ls', '-t', '/sdcard/*.mp4']
+            result = subprocess.run(cmd, capture_output=True, text=True, timeout=5)
```

Ruff (0.15.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)